haystack/nodes/extractor/entity.py

```diff
@@ -185,3 +509,360 @@ def simplify_ner_for_qa(output):
+                      (on languages that support that meaning, which is basically tokens separated by a space).
+                      These mitigations will only work on real words, "New york" might still be tagged with two different entities.
+            "first": Will use the SIMPLE strategy except that words, cannot end up with
+                     different tags. Words will simply use the tag of the first token of the word when there is ambiguity.
+            "average": Will use the SIMPLE strategy except that words, cannot end up with
+                       different tags. The scores will be averaged across tokens, and then the label with the maximum score is chosen.
+            "max": Will use the SIMPLE strategy except that words, cannot end up with
+                   different tags. Word entity will simply be the token with the maximum score.
+        :param ignore_labels: Optionally specify a list of labels to ignore. If None is specified it
+            defaults to `["O"]`.
+        """
+        if ignore_labels is None:
+            ignore_labels = ["O"]
+        logits = model_outputs["logits"][0].numpy()
+        sentence = model_outputs["sentence"]
+        input_ids = model_outputs["input_ids"][0]
+        offset_mapping = model_outputs["offset_mapping"][0].numpy()
+        special_tokens_mask = model_outputs["special_tokens_mask"][0].numpy()
+        word_ids = model_outputs["word_ids"]
+        word_offset_mapping = model_outputs.get("word_offset_mapping", None)
+
+        maxes = np.max(logits, axis=-1, keepdims=True)
+        shifted_exp = np.exp(logits - maxes)
+        scores = shifted_exp / shifted_exp.sum(axis=-1, keepdims=True)
+
+        updated_offset_mapping = offset_mapping
+        pre_entities = self.gather_pre_entities(
+            sentence, input_ids, scores, updated_offset_mapping, special_tokens_mask, word_ids
+        )
+        grouped_entities = self.aggregate(pre_entities, aggregation_strategy, word_offset_mapping=word_offset_mapping)
+        # Filter anything that is in self.ignore_labels
+        entities = [
+            entity
+            for entity in grouped_entities
+            if entity.get("entity", None) not in ignore_labels and entity.get("entity_group", None) not in ignore_labels
+        ]
+        return entities
+
+    def aggregate(
+        self, pre_entities: List[Dict[str, Any]], aggregation_strategy: str, word_offset_mapping: List[Tuple] = None
+    ) -> List[Dict[str, Any]]:
+        """Aggregate the `pre_entities` depending on the `aggregation_strategy`.
+
+        :param pre_entities: List of entity predictions for each token in a text.
+        :param aggregation_strategy: The strategy to fuse (or not) tokens based on the model prediction.
+        :param word_offset_mapping: List of (word, (char_start, char_end)) tuples for each word in a text.
+        """
+        if aggregation_strategy in {AggregationStrategy.NONE, AggregationStrategy.SIMPLE}:
```

Comment:
@sjrl perhaps it would be better to use these ExplicitEnums throughout rather than mix them and str?

Reply:
Ahh good catch. I meant to replace these with just the strings. The explicit enum comes from the transformers library and I don't think it would be good to depend on that in case they change it in the future.